src/EditorFeatures/CSharp/EventHookup/EventHookupCommandHandler_TabKeyCommand.cs

```diff
@@ -156,7 +158,8 @@ private void GenerateAndAddEventHandler(ITextView textView, ITextBuffer subjectB
 
             var documentWithNameAndAnnotationsAdded = AddMethodNameAndAnnotationsToSolution(document, eventHandlerMethodName, position, plusEqualsTokenAnnotation, cancellationToken);
             var semanticDocument = SemanticDocument.CreateAsync(documentWithNameAndAnnotationsAdded, cancellationToken).WaitAndGetResult(cancellationToken);
-            var updatedRoot = AddGeneratedHandlerMethodToSolution(semanticDocument, eventHandlerMethodName, plusEqualsTokenAnnotation, cancellationToken);
+            var preferences = CSharpCodeGenerationPreferences.FromDocumentAsync(semanticDocument.Document, cancellationToken).WaitAndGetResult(cancellationToken);
```

Comment:
interesting that it's Csharp/VB specific.  i'll have to wrap my head around that.  guessing that that's because of langversions for now.

Reply:
Currently it's just langversion, but if we take next step and replace OptionSet with explicit options all but of the used options are CSharp specific.